src/utils/pbconvert.hpp

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef PBCONVERT_HPP_
+#define PBCONVERT_HPP_
+
+#include <aos/common/types.hpp>
+
+#include <proto/servicemanager/v3/servicemanager.pb.h>
+
+/**
+ * Converts Aos string to PB string.
+ *
+ * @tparam cSize max PB string size.
+ * @param src Aos string to convert from.
+ */
+template <size_t cSize>
+constexpr void StringToPB(const aos::String& src, char (&dst)[cSize])
+{
+    aos::String(dst, cSize - 1) = src;
+}
+
+/**
+ * Converts PB string to Aos string.
+ *
+ * @tparam cSize max PB string size.
+ * @param dst Aos string to convert to.
+ */
+template <size_t cSize>
+constexpr void PBToString(const char (&src)[cSize], aos::String& dst)
+{
+    auto len = strlen(src);
+
+    if (len > cSize - 1) {
+        len = cSize - 1;
+    }
+
+    dst = aos::String(src, len);
+}
+
+/**
+ * Converts PB string to Aos enum.
+ *
+ * @tparam T Aos enum type.
+ * @tparam cSize max PB string size.
+ * @param dst Aos enum to convert to.
+ */
+
+template <typename T, size_t cSize>
+constexpr void PBToEnum(const char (&src)[cSize], T& dst)
+{
+    auto len = strlen(src);
+
+    if (len > cSize - 1) {
+        len = cSize - 1;
+    }
+
+    dst.FromString(aos::String(src, len));
```

Comment:
Taking into account FromString could fail either assert check should be used or function should return error.

Reply:
Yes, but I don't want to check error for pbconvert function, otherwise it will bloat the code drastically. Assert added.